app/controllers/hq/sponsorships_controller.rb

```diff
@@ -20,6 +33,23 @@ def destroy
 
   private
 
+  def create_sponsorship_for(sponsor, orphan)
+    if @sponsorship_creator.call
+      if sponsor.request_fulfilled
+        flash[:success] = 'Sponsorship link was successfully created.'
+        redirect_to hq_sponsor_path(sponsor)
+      else
+        flash[:success] = "Sponsorship link was successfully created for #{orphan.full_name}"
+        redirect_to hq_new_sponsorship_path(sponsor, scope: 'eligible_for_sponsorship')
+      end
+    end
+  end
+
+  def redirect_back_to_new_sponsorship_for(sponsor)
+    flash[:warning] = @sponsorship_creator.error_msg
```

Comment:
this should be `flash[:error]`
Also add some steps to the integration tests to check for sad paths too !